plugins/perms/perms.plugin.zsh

```diff
@@ -0,0 +1,78 @@
+# Some useful commands for setting permissions.
+#
+# Rory Hardy [GneatGeek]
+# Andrew Janke [apjanke]
+
+### Aliases
+
+# Set all files' permissions to 644 recursively in a directory
+alias set644='find . -type f -print0 | xargs -0 chmod 644'
+
+# Set all directories' permissions to 755 recursively in a directory
+alias set755='find . -type d -print0 | xargs -0 chmod 755'
+
+### Functions
+
+# fixperms - fix permissions on files and directories, with confirmation
+# Returns 0 on success, nonzero if any errors occurred
+fixperms () {
+  local opts confirm target exit_status chmod_opts use_slow_mode
+  zparseopts -E -D -a opts -help -slow v+=chmod_opts
+  if [[ $# > 1 || -n "${opts[(r)--help]}" ]]; then
+    cat <<EOF
+Usage: fixperms [-v] [--help] [--slow] [target]
```

Comment:
If you use tabs for indentation you can then use `cat <<-EOF` (notice the extra hyphen) to strip the tabs. You'd  then have to indent using spaces inside the heredoc, so it's not a clear benefit, just throwing it out there for your information. 😁